### PR TITLE
[backend] Fix #97

### LINKED ISSUE #97: [backend] Fix Buyer/ API issues
- [x] `GET /buyer/order` lack of thumbnail product's name
~~`GET /buyer/order/:id` lack of order status' timestamp~~
~~Current Schema is not recording order status changed time~~
- [x] `POST /buyer/checkout` failed to update `"product_archive"`

### PR DESCRIPTION
**Pull request by Error0229**
Resolves #97

### Changes

- Add `product_name` for `GET /buyer/order`
- Fix `product_archive`  not properly updating issue

### CODE REVIEW

**Comment by KatoToyoki** [comment]
i wonder how the four status will be handled. 
will it be that you give me the current status string, and the previous ones i need to handle by myself to recorded "done"?

**Comment by Error0229** [comment]
> i wonder how the four status will be handled. will it be that you give me the current status string, and the previous ones i need to handle by myself to recorded "done"?

Yes, there are 5 order status defined in db, it looks like
```sql
CREATE TYPE "order_status" AS ENUM (
    'paid',
    'shipped',
    'delivered',
    'cancelled',
    'finished'
);
```
You might need to create a class for that to handle status changes.

It is important that the next status of `delivered` can be either `cancelled` or `finished`

**Review by KatoToyoki — DISMISSED** [review]
ok understood

**Review by KatoToyoki — DISMISSED** [review]

File: db/queries/buyer.sql [hunk 1]
```diff
@@ -559,17 +584,24 @@ insert_archive AS (
...
-        P."description",
-        P."price",
-        P."image_id"
-    FROM
-        "product" AS P,
-        check_version AS CV
-    WHERE
-        P."id" = $1
-        AND CV."version_existed" = FALSE)
+        P."version" +(
```
Comment (x06lan) [inline db/queries/buyer.sql]: ?

Comment (Error0229) [reply db/queries/buyer.sql]: When the product doesn't exist in the product archive, it doesn't need to add 1 to its version.
Then, when the condition is true, the select query will return 1. Also, I added something to make it really work.

**Review by x06lan — APPROVED** [review]

**Review by KatoToyoki — APPROVED** [review]

**Pull request merged**